server/common/ConfigKVParser.java

```diff
@@ -226,7 +229,25 @@ boolean isNested() {
             public static final Leaf<Path> PATH = new Leaf<>(
                     (yaml) -> yaml.isString(),
                     (yaml) -> Paths.get(yaml.asString().value()),
-                    "<path>"
+                    "<relative or absolute path>"
+            );
+            public static final Leaf<Long> BYTES_SIZE = new Leaf<>(
+                    (yaml) -> yaml.isString() && Bytes.Parser.isValidSizeString(yaml.asString().value()),
+                    (yaml) -> Bytes.Parser.parse(yaml.asString().value()),
+                    "<size>"
+            );
+            public static final Leaf<InetSocketAddress> INET_SOCKET_ADDRESS = new Leaf<>(
+                    (yaml) -> {
+                        if (!yaml.isString()) return false;
+                        // use URI to parse IPV4, IVP4 and host names - however, we must add a dummy scheme to use it
+                        URI uri = URI.create("scheme://" + yaml.asString().value());
+                        return uri.getHost() != null && uri.getPort() != -1;
+                    },
+                    (yaml) -> {
+                        URI uri = URI.create("scheme://" + yaml.asString().value());
+                        return new InetSocketAddress(uri.getHost(), uri.getPort());
+                    },
+                    "<address:port>"
```

Comment:
we define two new types of configuration entries:
1. a `Size`, which uses the Bytes parser to read a user size entry such as "5 mb" into bytes
2. a `Address`, which parses things like `123.4.5.1:1123` or `web.blabla.com:1823`